### PR TITLE
COMPAT: Add back remove_na for seaborn

### LINKED ISSUE #16971: COMPAT: replace import of remove_na
https://travis-ci.org/pandas-dev/pandas/jobs/253985437

xref #16935

this breaks ``seaborn`` which is apparently using this private method.

So let's

- open an issue on seaborn about this (optionally they can simply use ``s[s.notnull()]`` instead of the import), prob would appreciate a PR
- add back the import to ``pandas.core.series`` and deprecate ``remove_na`` (which just can call ``remove_na_arraylike``

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

Closes #16971.

### CODE REVIEW

**Comment by gfyoung** [comment]
Can't use `deprecate` decorator because it introduces a circular import unfortunately.

**Comment by gfyoung** [comment]
FYI : couldn't use `deprecate` from `decorators` because that introduces a circular import 😢

File: pandas/core/dtypes/missing.py [hunk 1]
```diff
@@ -401,3 +403,16 @@ def remove_na_arraylike(arr):
     Return array-like containing only true/non-NaN values, possibly empty.
     """
     return arr[notnull(lib.values_from_object(arr))]
+
+
+# see gh-16971
+def remove_na(arr):
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: move this to pandas.core.series

File: pandas/tests/dtypes/test_missing.py [hunk 2]
```diff
@@ -14,7 +14,13 @@
 from pandas.core.dtypes.dtypes import DatetimeTZDtype
 from pandas.core.dtypes.missing import (
     array_equivalent, isnull, notnull,
-    na_value_for_dtype)
+    na_value_for_dtype, remove_na)
+
+
```
Comment (jreback) [inline pandas/tests/dtypes/test_missing.py]: to pandas/series/test_missing.py

File: pandas/core/dtypes/missing.py [hunk 3]
```diff
@@ -401,3 +403,16 @@ def remove_na_arraylike(arr):
...
+    """
+    DEPRECATED : this function will be removed in a future version.
+
+    Alias to `remove_na_arraylike` for `seaborn` compatibility.
+    """
+
+    warnings.warn("remove_na is deprecated. Use remove_na_arraylike instead.",
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: no recommendation at all. this is a private function.

File: pandas/core/dtypes/missing.py [hunk 4]
```diff
@@ -401,3 +403,16 @@ def remove_na_arraylike(arr):
     Return array-like containing only true/non-NaN values, possibly empty.
     """
     return arr[notnull(lib.values_from_object(arr))]
+
+
+# see gh-16971
+def remove_na(arr):
+    """
+    DEPRECATED : this function will be removed in a future version.
+
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: remove the 2nd sentence.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/dtypes/test_missing.py (hunk 2 above)
Comment (gfyoung) [inline pandas/tests/dtypes/test_missing.py]: Done.

File: pandas/core/dtypes/missing.py (hunk 3 above)
Comment (gfyoung) [inline pandas/core/dtypes/missing.py]: Fair enough.  Done.

File: pandas/core/dtypes/missing.py (hunk 4 above)
Comment (gfyoung) [inline pandas/core/dtypes/missing.py]: Done.

File: pandas/core/dtypes/missing.py (hunk 1 above)
Comment (gfyoung) [inline pandas/core/dtypes/missing.py]: Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**